File: python/src/reality_capture/specifications/fill_image_properties.py
from pydantic import BaseModel, Field
from typing import Optional
from enum import Enum


class FillImagePropertiesInputs(BaseModel):
    """
    List of possible inputs for a Fill Image Properties job.
    """
    image_collections: Optional[list[str]] = Field(default=None,
                                                   description="List of image collection reality data ids.",
                                                   alias="imageCollections")
    scene_to_process: Optional[str] = Field(default=None,
                                            description="Reality data id of ContextScene to process.",
                                            alias="sceneToProcess")
    scene_to_complete: Optional[str] = Field(default=None,
                                             description="Reality data id of ContextScene to complete.",
                                             alias="sceneToComplete")
    preset: Optional[str] = Field(default=None,
                                  description="Path to preset prefixed with reality data id.")


class FillImagePropertiesOutputs(BaseModel):
    """
    List of possible outputs for a Fill Image Properties job.
    """
    scene: str = Field(description="Output reality data id of ContextScene with image properties filled.")


class AltitudeReference(Enum):
    """
    Values for Altitude Reference when filling image properties.
    """
    SEA_LEVEL = "SeaLevel"
    WGS84_ELLIPSOID = "WGS84Ellipsoid"


class FillImagePropertiesOptions(BaseModel):
    """
    List of possible options for Fill Image Properties job.
    """
    recursive_image_collections: Optional[bool] = Field(default=None,
                                                        description="Recursively read folders in image collection.",
                                                        alias="recursiveImageCollections")
    altitude_reference: Optional[AltitudeReference] = Field(default=None,
                                                            description="Reference altitude when reading Z data "
                                                                        "from Exif.",
                                                            alias="altitudeReference")


class FillImagePropertiesSpecifications(BaseModel):
    """
    Specifications for Fill Image Properties job.
    """
    inputs: FillImagePropertiesInputs = Field(description="Inputs for Fill Image Properties job")
    outputs: FillImagePropertiesOutputs = Field(description="Outputs for Fill Image Properties job")
    options: Optional[FillImagePropertiesOptions] = Field(None, description="Options for Fill Image Properties job")


class FillImagePropertiesOutputsCreate(Enum):
    """
    Values for outputs for Fill Image Properties job creation.
    """
    SCENE = "scene"


class FillImagePropertiesSpecificationsCreate(BaseModel):
    """
    Specifications for Fill Image Properties job creation.
    """
    inputs: FillImagePropertiesInputs = Field(description="Inputs for Fill Image Properties job")
    outputs: list[FillImagePropertiesOutputsCreate] = Field(description="List of output type for Fill "
                                                                        "Image Properties job")
    options: Optional[FillImagePropertiesOptions] = Field(None, description="Outputs for Fill Image Properties job")

File: typescript_sdk/reality_capture/src/specifications/fill_image_properties.ts
import { z } from "zod";

export enum AltitudeReference {
  SEA_LEVEL = "SeaLevel",
  WGS84_ELLIPSOID = "WGS84Ellipsoid",
}

export const FillImagePropertiesInputsSchema = z.object({
  imageCollections: z.array(z.string()).optional().describe("List of image collection reality data ids."),
  sceneToProcess: z.string().optional().describe("Reality data id of ContextScene to process."),
  sceneToComplete: z.string().optional().describe("Reality data id of ContextScene to complete."),
  presets: z.array(z.string()).optional().describe("List of paths to preset"),
});
export type FillImagePropertiesInputs = z.infer<typeof FillImagePropertiesInputsSchema>;

export const FillImagePropertiesOutputsSchema = z.object({
  scene: z.string().describe("Output reality data id of ContextScene with image properties filled."),
});
export type FillImagePropertiesOutputs = z.infer<typeof FillImagePropertiesOutputsSchema>;

export const FillImagePropertiesOptionsSchema = z.object({
  recursiveImageCollections: z.boolean().optional()
    .describe("Recursively read folders in image collection."),
  altitudeReference: z.nativeEnum(AltitudeReference).optional()
    .describe("Reference altitude when reading Z data from Exif."),
});
export type FillImagePropertiesOptions = z.infer<typeof FillImagePropertiesOptionsSchema>;

export const FillImagePropertiesSpecificationsSchema = z.object({
  inputs: FillImagePropertiesInputsSchema
    .describe("Inputs for Fill Image Properties job"),
  outputs: FillImagePropertiesOutputsSchema
    .describe("Outputs for Fill Image Properties job"),
  options: FillImagePropertiesOptionsSchema.optional()
    .describe("Options for Fill Image Properties job"),
});
export type FillImagePropertiesSpecifications = z.infer<typeof FillImagePropertiesSpecificationsSchema>;

export enum FillImagePropertiesOutputsCreate {
  SCENE = "scene",
}

export const FillImagePropertiesSpecificationsCreateSchema = z.object({
  inputs: FillImagePropertiesInputsSchema
    .describe("Inputs for Fill Image Properties job"),
  outputs: z.array(z.nativeEnum(FillImagePropertiesOutputsCreate))
    .describe("List of output type for Fill Image Properties job"),
  options: FillImagePropertiesOptionsSchema.optional()
    .describe("Outputs for Fill Image Properties job"),
});
export type FillImagePropertiesSpecificationsCreate = z.infer<typeof FillImagePropertiesSpecificationsCreateSchema>;

export const FillImagePropertiesCostSchema = z.object({
  imageCount: z.number().describe("Number of images to import."),
});
export type FillImagePropertiesCost = z.infer<typeof FillImagePropertiesCostSchema>;